modules/common-fs2/src/main/scala/com/snowplowanalytics/snowplow/enrich/common/fs2/Environment.scala

```diff
@@ -252,4 +262,25 @@ object Environment {
       case _ =>
         None
     }
+
+  /**
+   * Sets up the Remote adapters for the ETL
+   * @param remoteAdaptersConfig List of configuration per remote adapter
+   * @return An Http4sClient with a mapping of vendor-version and the adapter assigned for it
+   */
+  def prepareRemoteAdapters[F[_]: ConcurrentEffect](
+    remoteAdaptersConfig: Option[RemoteAdapterConfigs],
+    ec: ExecutionContext
+  ): Resource[F, (Http4sClient[F], Map[(String, String), RemoteAdapter])] =
+    remoteAdaptersConfig.fold(
+      Clients.mkHttp(None, None, None, ec).map((_, Map.empty[(String, String), RemoteAdapter]))
+    ) { adapters =>
```

Comment:
This is an example of some code that is hard to understand, but would be much easier to understand if `RemoteAdapterConfigs` is non-optional.